scipy/stats/_discrete_distns.py

```diff
@@ -152,6 +152,86 @@ def _entropy(self, p):
 bernoulli = bernoulli_gen(b=1, name='bernoulli')
 
 
+class betabinom_gen(rv_discrete):
+    r"""A beta-binomial discrete random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The beta-binomial distribution is a binomial distribution with a
+    probability of success `p` that follows a beta distribution.
+
+    The probability mass function for `betabinom` is:
+
+    .. math::
+
+       f(k) = \binom{n}{k} \frac{B(k + a, n - k + b)}{B(a, b)}
+
+    for ``k`` in ``{0, 1,..., n}``, where :math:`B(a, b)` is the beta function.
+
+    `betabinom` takes :math:`n`, :math:`a`, and :math:`b` as shape parameters.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Beta-binomial_distribution
+
+    %(after_notes)s
+
+    .. versionadded:: 1.4.0
+
+    See Also
+    --------
+    beta, binom
+
+    %(example)s
+
+    """
+
+    def _rvs(self, n, a, b):
+        p = self._random_state.beta(a, b, self._size)
+        return self._random_state.binomial(n, p, self._size)
+
+    def _get_support(self, n, a, b):
+        return 0, n
+
+    def _argcheck(self, n, a, b):
+        return (n >= 0) & (a > 0) & (b > 0)
```

Comment:
one could even say n > 0, n=0 is a degenerate case, not sure if that could be an issue

Reply:
It is a degenerate case, but I tried creating an instance with `n = 0` and found that `.rvs()`, `.mean()`, `.median()`, etc. return correct values. `.ppf(0)` returns `-1`, which is nonsense.

Reply:
The closely related `scipy.stats.binom` distribution does permit `n = 0`, and it also has the issue that `binom(0, p).ppf(0)` returns `-1`. I think that for the sake of consistency, it's best if `betabinom` also accepts `n = 0`. If this is perceived as an actual bug, then it would be best to change the behavior of both distributions for this degenerate case in a separate pull request.

Reply:
Ok, this case is not really important and I am fine to leave it for consistency for now